service/src/main/java/com/derpgroup/derpwizard/dao/impl/AccountLinkingDAOFactory.java

```diff
@@ -0,0 +1,19 @@
+package com.derpgroup.derpwizard.dao.impl;
+
+import com.derpgroup.derpwizard.configuration.AccountLinkingDAOConfig;
+import com.derpgroup.derpwizard.dao.AccountLinkingDAO;
+
+public class AccountLinkingDAOFactory {
+
+  public static AccountLinkingDAO getDAO(AccountLinkingDAOConfig config){
+    AccountLinkingDAO dao = null;
+    switch(config.getType().toUpperCase()){
+    case "H2": 
```

Comment:
This is a factory, so we should really just be properly doing dependency injection, but for now H2 is the only value that can be used here.